core/server/web/parent-app.js

```diff
@@ -37,17 +43,31 @@ module.exports = function setupParentApp(options = {}) {
     // This sets global res.locals which are needed everywhere
     parentApp.use(shared.middlewares.ghostLocals);
 
+    // Wrap the admin and API apps into a single express app for use with vhost
+    const adminApp = express();
+    adminApp.enable('trust proxy'); // required to respect x-forwarded-proto in admin requests
```

Comment:
Took me far too long to realise this was why I was having major issues with redirects 😞 I was surprised it was needed though, I expected the `req.secure` to be set on the parent app before it's passed through to the child app. We use the `app = express()` pattern for the API versions too which was why I didn't think of trying this earlier.